nerfstudio/scripts/datasets/process_project_aria.py

```diff
@@ -0,0 +1,227 @@
+    nearest_pose_idx = np.searchsorted(t_world_devices.timestamps_ns, capture_time_ns)
+    nearest_pose_idx = np.minimum(nearest_pose_idx, len(t_world_devices.timestamps_ns) - 1)
+    assert nearest_pose_idx != -1, f"Could not find pose for {capture_time_ns}"
+    t_world_device = t_world_devices.t_world_devices[nearest_pose_idx]
+
+    # Compute the world to camera transform.
+    t_world_camera = t_world_device @ camera_calibration.t_device_camera @ T_ARIA_NERFSTUDIO
+
+    return AriaImageFrame(
+        camera=camera_calibration,
+        file_path=file_path,
+        t_world_camera=t_world_camera,
+        timestamp_ns=capture_time_ns,
+    )
+
+
+def to_nerfstudio_frame(frame: AriaImageFrame) -> Dict:
+    return {
+        "fl_x": frame.camera.fx,
+        "fl_y": frame.camera.fy,
+        "cx": frame.camera.cx,
+        "cy": frame.camera.cy,
+        "distortion_params": frame.camera.distortion_params.tolist(),
+        "w": frame.camera.width,
+        "h": frame.camera.height,
+        "file_path": frame.file_path,
+        "transform_matrix": frame.t_world_camera.to_matrix().tolist(),
+        "timestamp": frame.timestamp_ns,
+    }
+
+
+@dataclass
+class ProcessProjectAria:
+    """Processes Project Aria data i.e. a VRS of the raw recording streams and the MPS attachments
+    that provide poses, calibration, and 3d points. More information on MPS data can be found at:
+      https://facebookresearch.github.io/projectaria_tools/docs/ARK/mps.
+    """
+
+    vrs_file: Path
+    """Path to the VRS file."""
+    mps_data_dir: Path
+    """Path to Project Aria Machine Perception Services (MPS) attachments."""
+    output_dir: Path
+    """Path to the output directory."""
+
+    def main(self) -> None:
+        """Generate a nerfstudio dataset from ProjectAria data (VRS) and MPS attachments."""
+        # Create output directory if it doesn't exist.
+        self.output_dir.mkdir(parents=True, exist_ok=True)
+
+        provider = create_vrs_data_provider(self.vrs_file.absolute().as_posix())
+        assert provider is not None, "Cannot open file"
+
+        name_to_camera = get_camera_calibs(provider)
+
+        print("Getting poses from closed loop trajectory CSV...")
+        trajectory_csv = self.mps_data_dir / "closed_loop_trajectory.csv"
+        t_world_devices = read_trajectory_csv_to_dict(trajectory_csv.absolute().as_posix())
+
+        name = "camera-rgb"
+        stream_id = provider.get_stream_id_from_label(name)
+
+        # create an AriaImageFrame for each image in the VRS.
+        print("Creating Aria frames...")
+        aria_frames = [
+            to_aria_image_frame(provider, index, name_to_camera, t_world_devices, self.output_dir)
+            for index in range(0, provider.get_num_data(stream_id))
+        ]
+
+        # create the NerfStudio frames from the AriaImageFrames.
+        print("Creating NerfStudio frames...")
+        RGB_VALID_RADIUS = 707.5
```

Comment:
> (1) Is this necessary to hardcode?
 Right now the project aria tools don't provide a way to obtain the valid mask directly as a mask image. So I think that leaves us with two options 1) stick with hardcoding this value in the script and sample pixels uniformly in a circle given this value (current solution) 2) generate the valid mask image for all exported images and then use the pixel sampler with masks. Happy to go for either solution, the only main advantage of option 1) I see is saving a lot of memory by not having to load/store/process masks for each image.

2) Indeed, this value needs to be scaled based on the recording profile. This value works for profiles with 1408x1408 RGB so I will update the code to detect what size RGB images are present in the VRS to accomodate different profiles.

Reply:
@sweeneychris 1415 would be the value for full image size. Since we have only two RGB images size, the best could be to check the ImageSize of the RGB stream and just set 707.5 or 1408 accordingly